lib/compat/wordpress-6.5/block-bindings/post-meta.php

```diff
@@ -15,17 +15,30 @@
  */
 function gutenberg_block_bindings_post_meta_callback( $source_attrs, $block_instance ) {
 	if ( empty( $source_attrs['key'] ) ) {
-		return null;
+		return '';
```

Comment:
If the configuration for the block binding is incorrect, then we probably should return `null` to skip processing. In general, when returning `null`, the original value will stay in the block's saved HTML:

https://github.com/WordPress/gutenberg/blob/702f78adc17295bef1402c7577c07ee4aa6ba028/lib/compat/wordpress-6.5/blocks.php#L202-L205

https://github.com/WordPress/wordpress-develop/blob/716cf66fda28643c7178ac43ffe5d27f239a493b/src/wp-includes/class-wp-block.php#L271-L274

In other cases, we could replace the value with an empty string. It's really hard to tell what is the best way forward.

Reply:
Feels to me like we should keep the block attribute value as is for all the use-cases where the field is unavailable or protected or anything.

Reply:
I'm not opposed to the idea. It would have certain implications, like we would have to come up with a good strategy for providing the fallback value serialized in the saved block. I raised a very similar concern in https://github.com/WordPress/gutenberg/pull/58895#issuecomment-1966341566 where the proposed implementation tries to keep in sync the external data with the block attribute. In that case, the fallback would be the last value present in the external source and it would remain the fallback value. So definitely, we should look at it all holistically.

Reply:
I agree it makes sense to keep it as a fallback value. I'll make the changes to adapt for that.